packages/react-i18n/src/i18n.ts

```diff
@@ -299,6 +363,31 @@ export class I18n {
     return WEEK_START_DAYS.get(country) || DEFAULT_WEEK_START_DAY;
   }
 
+  // Intl.NumberFormat sometimes annotates the "curerncy symbol" with a country code.
+  // For example, in locale 'fr-FR', 'USD' is given the "symbol" of " $US".
+  // This method strips out the country-code annotation, if there is one.
+  // (So, for 'fr-FR' and 'USD', the return value would be " $").
+  //
+  // For other currencies, e.g. CHF and OMR, the "symbol" is the ISO currency code.
+  // In those cases, we return the full currency code without stripping the country.
+  getShortCurrencySymbol(currencyCode?: string) {
+    const currency = currencyCode || this.defaultCurrency || '';
+    const regionCode = currency.substring(0, 2);
```

Comment:
This seems really prone to being not true in some instances to me, is there something that makes it a guarantee that all currency codes will work this way?

Reply:
Well, it is the way ISO creates their codes; first two letters are the region (usually the country, but there are some non-country regions, e.g. `EUR` and `XAF`, for currencies that get used in more than one country.).
https://en.wikipedia.org/wiki/ISO_4217#Code_formation